frontend/src/components/stages/reveal_editor.ts

```diff
@@ -44,25 +42,58 @@ export class RevealEditor extends MobxLitElement {
         <div class="left">Stages to reveal</div>
         ${this.renderAddMenu()}
       </div>
-      ${this.stage.items.map(
-        (item, index) => this.renderRevealStageItem(item, index)
+      ${this.stage.items.map((item, index) =>
+        this.renderRevealStageItem(item, index)
       )}
     `;
   }
 
   private renderAddMenu() {
     if (!this.stage) return nothing;
 
-    const stageOptions = getStagesWithReveal(this.experimentEditor.stages).filter(
-      stage => stage.id !== this.stage?.id
-    );
+    // Get all stages with a reveal stage.
+    // Get all stages with a reveal stage.
+    const getStageOptions = () => {
+      const revealStages = getStagesWithReveal(
+        this.experimentEditor.stages
+      ).filter((stage) => stage.id !== this.stage?.id);
+
+      // Get the current stage index.
+      const currentStageIndex = this.experimentEditor.stages.findIndex(
+        (stage) => stage.id === this.stage!.id
+      );
+
+      // Filter stages to only those before the current stage index.
+      const filteredStages = revealStages.filter((stage) => {
+        const stageIndex = this.experimentEditor.stages.findIndex(
+          (expStage) => expStage.id === stage.id
+        );
+        return stageIndex < currentStageIndex;
+      });
+
+      return filteredStages;
+    };
+
+    const stageOptions = getStageOptions();
+    if (stageOptions.length === 0) {
```

Comment:
Instead of rendering twice, can we wrap the tooltip around the main return statement and set tooltip text to empty (which means no tooltip will render) if disabled?

Reply:
Done.